spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/actions/TestRemoveOrphanFilesAction.java

```diff
@@ -955,4 +956,114 @@ protected long waitUntilAfter(long timestampMillis) {
+            ImmutableMap.of(),
+                ImmutableMap.of(),
+            DeleteOrphanFiles.PrefixMismatchMode.ERROR));
+
+    Map<String, String> equalSchemes = Maps.newHashMap();
+    equalSchemes.put("s3n", "s3");
+    equalSchemes.put("s3a", "s3");
+    executeTest(validFiles,
+        actualFiles,
+        Lists.newArrayList(),
+        equalSchemes,
+            ImmutableMap.of(),
+        DeleteOrphanFiles.PrefixMismatchMode.ERROR);
+  }
+
+  @Test
+  public void testPathsWithEqualAuthorities() {
+    List<String> validFiles = Lists.newArrayList("hdfs://servicename1/dir1/dir2/file1");
+    List<String> actualFiles = Lists.newArrayList("hdfs://servicename2/dir1/dir2/file1");
+    AssertHelpers.assertThrows("Test remove orphan files with equal authorities",
+        ValidationException.class,
+        "Conflicting authorities/schemes: [(servicename1, servicename2)]",
+        () -> executeTest(validFiles,
+            actualFiles,
+            Lists.newArrayList(),
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+            DeleteOrphanFiles.PrefixMismatchMode.ERROR));
+
+    Map<String, String> equalAuthorities = Maps.newHashMap();
+    equalAuthorities.put("servicename1", "servicename");
+    equalAuthorities.put("servicename2", "servicename");
+    executeTest(validFiles,
+        actualFiles,
+        Lists.newArrayList(),
+            ImmutableMap.of(),
+        equalAuthorities,
+        DeleteOrphanFiles.PrefixMismatchMode.ERROR);
+  }
+
+  @Test
+  public void testRemoveOrphanFileActionWithDeleteMode() {
+    List<String> validFiles = Lists.newArrayList("hdfs://servicename1/dir1/dir2/file1");
+    List<String> actualFiles = Lists.newArrayList("hdfs://servicename2/dir1/dir2/file1");
+
+    executeTest(validFiles,
+        actualFiles,
+        Lists.newArrayList("hdfs://servicename2/dir1/dir2/file1"),
+            ImmutableMap.of(),
+            ImmutableMap.of(),
+        DeleteOrphanFiles.PrefixMismatchMode.DELETE);
+  }
+
+  private void executeTest(List<String> validFiles,
+                           List<String> actualFiles,
+                           List<String> expectedOrphanFiles) {
+    executeTest(validFiles, actualFiles, expectedOrphanFiles, ImmutableMap.of(), ImmutableMap.of(),
+        DeleteOrphanFiles.PrefixMismatchMode.IGNORE);
+  }
+
+  private void executeTest(List<String> validFiles,
+                           List<String> actualFiles,
+                           List<String> expectedOrphanFiles,
+                           Map<String, String> equalSchemes,
+                           Map<String, String> equalAuthorities,
+                           DeleteOrphanFiles.PrefixMismatchMode mode) {
+    Dataset<Row> validFilesDF = spark.createDataset(validFiles, Encoders.STRING()).toDF();
+    Dataset<Row> actualFilesDF = spark.createDataset(actualFiles, Encoders.STRING()).toDF();
+
+    List<String> orphanFiles = BaseDeleteOrphanFilesSparkAction.findOrphanFiles(
```

Comment:
nit: We can use `DeleteOrphanFilesSparkAction` now

Reply:
done